Add TestGrid Documentation and Linting (#10)
CSPI-QE/ocp-ci-docs: docs/OCP_CI_Tutorials/Scenarios/Scenarios_Guide.md

## Overview

The following Headers are phases that MUST be done for each scenario. They are needed to provide a structure to the many OpenShift layered product scenarios that will be running on OpenShift CI.

> Below is a basic flow chart of how a OpenShift CI scenario may look. Each scenario should have a chain where most of the work is completed. This chain should be named something like `interop-{product_name}-chain`. The scenario chain should execute the orchestrate, execute, and report steps in order. The orchestrate, execute, and report steps of a scenario's chain can either be another chain or a single ref depending on your scenario's specific needs.

Reviewer: ```suggestion
Below is a basic flow chart of how an OpenShift CI scenario may look. Each scenario should have a chain where most of the work is completed. This chain should be named something like `interop-{product_name}-chain`. The scenario chain should execute the orchestrate, execute, and report steps in order. The orchestrate, execute, and report steps of a scenario's chain can either be another chain or a single ref depending on your scenario's specific needs.
```

Author: Resolved in [**39fe241**7dca0aa2df5e3e0cb4676f4fa7ac99bf7](https://github.com/CSPI-QE/ocp-ci-docs/pull/10/commits/39fe2417dca0aa2df5e3e0cb4676f4fa7ac99bf7)

  end

```

## Provision

We define provision as:

> > The initial setup of your scenarios files and the OpenShift installation

Reviewer: ```suggestion
> The initial setup of your scenario's files and the OpenShift installation
```

Author: Resolved in [**39fe241**7dca0aa2df5e3e0cb4676f4fa7ac99bf7](https://github.com/CSPI-QE/ocp-ci-docs/pull/10/commits/39fe2417dca0aa2df5e3e0cb4676f4fa7ac99bf7)

See the [Orchestration Guide](Orchestration_Guide.md)) for detailed information meant to help you organize the installation of your product.

## Execute

We define execute as:

> The execution of any interop tests in a scenario. This step should only consist of any setup that absolutely must occur in this step (rather than the [orchestration step](Orchestration_Guide.md)) and the execution of interop tests. The execution of these tests should result in one or more JUnit (xml) files.

See the [Execution Guide](Execution_Guide.md) for detailed information meant to help you organize your test execution.

## Report

TBD

See the [Reporting Guide](Reporting_Guide.md) for detailed information meant to help you report your test execution.

## Document

TBD

See the [Scenario Documentation Guide](../../Policy/Documentation/Scenario_Documentation_Policy.md) for detailed information meant to help you document your scenario.

## Production

This is meant to be a sort of final check just to make sure that you and your reviewers are aware of what needs to be done to push to production (a.k.a merging the PR).

### Remove Spot Instances

> We cannot merge any code into the release repo meant to be used in production that is using Spot Instances. These can be unreliable and will lead to clusters going offline during production runs. We should be using them to save costs in development but when it is time to merge the scenario code we must ensure we remove the use of spot instances.

Reviewer: ```suggestion
We cannot merge any code into the release repo meant to be used in production that is using spot instances. These can be unreliable and will lead to clusters going offline during production runs. We should be using them to save costs in development, but when it is time to merge the scenario code we must ensure we remove the use of spot instances.
```

Author: Resolved in [**39fe241**7dca0aa2df5e3e0cb4676f4fa7ac99bf7](https://github.com/CSPI-QE/ocp-ci-docs/pull/10/commits/39fe2417dca0aa2df5e3e0cb4676f4fa7ac99bf7)

      BASE_DOMAIN: aws.interop.ccitredhat.com
    workflow: cucushift-installer-rehearse-aws-ipi
```

> A good way to verify your using what you expect to use for the value of cron you can use [Cronitor](https://crontab.guru/#0_1_*_*_1)

Reviewer: ```suggestion
A good way to verify that you're using what you expect to use for the value of cron, you can use [Cronitor](https://crontab.guru/#0_1_*_*_1)
```

Author: Resolved in [**39fe241**7dca0aa2df5e3e0cb4676f4fa7ac99bf7](https://github.com/CSPI-QE/ocp-ci-docs/pull/10/commits/39fe2417dca0aa2df5e3e0cb4676f4fa7ac99bf7)

### Ensure Test execution output matches reporting

In order for us to merge a MR we need to be sure the E2E testing is working. This includes reporting the test results.

It is important that we verify the different levels of artifacts and make sure that they all align.

For example, we might need to check the following:

> - Console output (prow job)

Reviewer: ```suggestion
- Console output (Prow job)
```

Author: Resolved in [**39fe241**7dca0aa2df5e3e0cb4676f4fa7ac99bf7](https://github.com/CSPI-QE/ocp-ci-docs/pull/10/commits/39fe2417dca0aa2df5e3e0cb4676f4fa7ac99bf7)
